jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -117,21 +118,13 @@ public String getOperationName() {
   @Override
   public Span setBaggageItem(String key, String value) {
     synchronized (this) {
-      // TODO emit a metric whenever baggage is updated
-      String prevItem = this.getBaggageItem(key);
-      this.context = this.context.withBaggageItem(key, value);
-      if (context.isSampled()) {
-        Map<String, String> fields = new HashMap<String, String>();
-        fields.put("event", "baggage");
-        fields.put("key", key);
-        fields.put("value", value);
-        if (prevItem != null) {
-          fields.put("override", "true");
-        }
-        return this.log(fields);
+      if (key == null || value == null) {
+        return this;
       }
+      BaggageValidity baggageValidity = this.getTracer().getBaggageRestrictionManager().isBaggageValid(key, value);
```

Comment:
how about 
  * s/BaggageValidity/BaggageRestriction/
  * s/isBaggageValue(key, value)/getBaggageRestriction(key)/
  * s/sanitizeBaggage/setBaggage/
  * move the call to getBaggageRestriction outside of `synchronized` since it's not dependent on the span